lib/descriptor/common.js

```diff
@@ -0,0 +1,273 @@
+/*!
+ * common.js - common functions for descriptor
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const {types} = require('../script/common');
+const BN = require('bcrypto/lib/bn.js');
+const common = exports;
+
+/**
+ * parse script context for descriptor
+ * @const {Object}
+ */
+
+common.scriptContext = {
+  TOP: 'TOP',
+  P2SH: 'P2SH',
+  P2WPKH: 'P2WPKH',
+  P2WSH: 'P2WSH'
+};
+
+/**
+ * Derive type for a key object in a descriptor
+ * @const {Object}
+ */
+
+common.deriveType = {
+  NO: 'NO',
+  UNHARDENED: 'UNHARDENED',
+  HARDENED: 'HARDENED'
+};
+
+/**
+ * String type for key object in a descriptor
+ * @const {Object}
+ */
+
+common.stringType = {
+  PUBLIC: 'PUBLIC',
+  PRIVATE: 'PRIVATE',
+  NORMALIZED: 'NORMALIZED'
+};
+
+/**
+ * Descriptor types
+ * @const {Object}
+ */
+
+common.descriptorTypes = {
+  PK: 'pk',
+  PKH: 'pkh',
+  WPKH: 'wpkh',
+  SH: 'sh',
+  WSH: 'wsh',
+  COMBO: 'combo',
+  ADDR: 'addr',
+  MULTI: 'multi',
+  RAW: 'raw'
+};
+
+/**
+ * Output types for descriptor
+ * @const {Object}
+ */
+
+common.outputTypes = types;
```

Comment:
why are we re-exporting `types` as `outputTypes`, why not just import from `lib/script/common.js`wherever `outputTypes` is used.